lib/merge.js

```diff
@@ -35,6 +36,17 @@ async function merge(context, pullRequest) {
     return false;
   }
 
+  if (mergeCommitMessageRegex) {
+    // If we find the regex, use the first capturing subgroup as new body (discarding whitespace).
+    const m = new RegExp(mergeCommitMessageRegex, 'sm').exec(pullRequest.body);
+    if (m) {
+      if (!m[1]) {
```

Comment:
@rix0rrr `!m[1]` will also be true for `""` (empty string), but I think we should allow that (the trim() later could produce an empty string anyways)